### PR TITLE
#254: qualify `std` namespace

### LINKED ISSUE #254: qualify `std::` namespace
Since we are currently embedding in namespace `std::` we make assumptions that standard library entities don't need to be qualified with `std::`. In order to move away from namespace `std::` we need to change that.

### PR DESCRIPTION
**Pull request by nmm0**
Closes #254

I honestly think I caught all of it. This builds if I change the mdspan namespace on gcc12 in C++14, 17, 20, and 23 modes. If I missed anything I'll catch it in the PR for changing the namespace...

### CODE REVIEW

File: include/experimental/__p0009_bits/mdspan.hpp [hunk 1]
```diff
@@ -324,6 +324,7 @@ class mdspan
   friend constexpr void swap(mdspan& x, mdspan& y) noexcept {
     // can't call the std::swap inside on HIP
     #if !defined(_MDSPAN_HAS_HIP) && !defined(_MDSPAN_HAS_CUDA)
+    using std::swap;
```
Comment (nmm0) [inline include/experimental/__p0009_bits/mdspan.hpp:327]: Note that this line was necessary for adl when the namespaces were renamed.

File: include/experimental/__p2630_bits/submdspan_extents.hpp [hunk 2]
```diff
@@ -14,6 +14,8 @@
 //
 //@HEADER
 
+#include <tuple>
```
Comment (nmm0) [inline include/experimental/__p2630_bits/submdspan_extents.hpp:17]: To be honest, I don't know why this suddenly needed to be added but it's necessary or clang frontend compilers fail

File: include/experimental/__p0009_bits/default_accessor.hpp [hunk 3]
```diff
@@ -35,20 +35,20 @@ struct default_accessor {
...
+      _MDSPAN_TRAIT(std::is_convertible, OtherElementType(*)[], element_type(*)[])
     )
   )
   MDSPAN_INLINE_FUNCTION
   constexpr default_accessor(default_accessor<OtherElementType>) noexcept {}
 
   MDSPAN_INLINE_FUNCTION
   constexpr data_handle_type
-  offset(data_handle_type p, size_t i) const noexcept {
+  offset(data_handle_type p, std::size_t i) const noexcept {
```
Comment (nliber) [inline include/experimental/__p0009_bits/default_accessor.hpp]: `size_t` is required to be in the global namespace due to C compatibility.

That being said, this doesn't hurt, so it is up to your discretion.

File: include/experimental/__p2630_bits/submdspan_extents.hpp (hunk 2 above)
Comment (nliber) [reply include/experimental/__p2630_bits/submdspan_extents.hpp:17]: `tuple` is used.  Even if it was accidentally included via another system header, this is a better change.

**Review by nliber — APPROVED** [review]

File: include/experimental/__p0009_bits/default_accessor.hpp (hunk 3 above)
Comment (nmm0) [reply include/experimental/__p0009_bits/default_accessor.hpp]: I see -- I kind of prefer it with the namespace but can remove it if people hate it.

Comment (crtrott) [reply include/experimental/__p0009_bits/default_accessor.hpp]: lets remove it again, makes for less change by a lot. just sed replace std::size_t with size_t :-)

**Review by crtrott — CHANGES_REQUESTED** [review]
Yeah lets revert std::size_t to size_t, we generally also have been using int64_t etc. unqualified in Kokkos for example. Its just a lot of extra characters.

**Review by crtrott — APPROVED** [review]

**Pull request merged**